packages/components/src/Filters/FilterDropdown.tsx

```diff
@@ -0,0 +1,103 @@
+import React from 'react';
+import { Dropdown, DropdownToggle, Level, DropdownProps } from '@patternfly/react-core';
+import FilterInput from './FilterInput';
+import './filter-dropdown.scss';
+
+export type FilterCategory = {
+  title: string;
+  type: string;
+  urlParam: string;
+  values: {
+    label: React.ReactNode;
+    value: string;
+  }[];
+};
+
+export interface FilterDropdownProps extends Omit<DropdownProps, 'onSelect'> {
+  addFilter?: (filterName?: string, selectedValue?: string | number, type?: string) => void;
+  removeFilter?: (filterName?: string, selectedValue?: string | number) => void;
+  hideCategories?: [string];
+  isOpen: boolean;
+  onToggle: (isOpen: boolean) => void;
+  filters?: Record<string, string>;
+  filterCategories?: FilterCategory[];
+  onSelect: ((event?: React.SyntheticEvent<HTMLDivElement, Event> | undefined) => void) | undefined;
+  label: string;
+  ouiaId: string;
+  ouiaSafe: boolean;
+}
+
+const FilterDropdown: React.FunctionComponent<FilterDropdownProps> = ({
+  hideCategories,
+  filters,
+  filterCategories,
+  onSelect,
+  label,
+  ouiaId,
+  ouiaSafe,
+  addFilter,
+  removeFilter,
+  isOpen = false,
+  onToggle,
+  ...props
+}) => {
+  const ouiaStateId = 'RHI/FilterDropdown';
+
+  const addRemoveFilters = (selectedValue?: string | number, filterName?: string, type?: string, isChecked?: boolean) => {
+    switch (type) {
+      case 'checkbox':
+        isChecked ? addFilter && addFilter(filterName, selectedValue, type) : removeFilter && removeFilter(filterName, selectedValue);
+        break;
+      case 'radio':
+        addFilter && addFilter(filterName, selectedValue, type);
+        break;
+    }
+  };
+
+  const ouiaFinalId = ouiaId !== undefined ? ouiaId : ouiaStateId;
+
+  return (
+    <Dropdown
+      className="ins-c-filter__dropdown"
+      {...props}
+      onSelect={onSelect}
+      toggle={
+        <DropdownToggle ouiaId={ouiaFinalId} ouiaSafe={ouiaSafe} onToggle={onToggle}>
+          {label}
+        </DropdownToggle>
+      }
+      isOpen={isOpen}
+      ouiaId={ouiaFinalId}
+      ouiaSafe={ouiaSafe}
+    >
+      <div className="pf-c-dropdown__menu-item">
+        {filterCategories &&
+          filterCategories.map(
+            (data: FilterCategory, index) =>
+              hideCategories &&
+              !hideCategories.includes(data.urlParam) && (
+                <Level key={`${data.urlParam}${index}`}>
+                  {data.title}
+                  {data.values.map((item: { label: React.ReactNode; value: string }, key: string | number) => (
```

Comment:
Let's not access the properties directly, if they are not passed down properly the component will blow up.
```suggestion
                  {data.title || ''}
                  {data.values?.map((item: { label: React.ReactNode; value: string } = {}, key: string | number) => (
```